keanu-project/src/main/java/io/improbable/keanu/distributions/continuous/Cauchy.java

```diff
@@ -0,0 +1,62 @@
+package io.improbable.keanu.distributions.continuous;
+
+import io.improbable.keanu.distributions.ContinuousDistribution;
+import io.improbable.keanu.distributions.dual.Diffs;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+import static io.improbable.keanu.distributions.dual.Diffs.L;
+import static io.improbable.keanu.distributions.dual.Diffs.S;
+import static io.improbable.keanu.distributions.dual.Diffs.X;
+
+public class Cauchy implements ContinuousDistribution {
+
+    private static final double NEG_LOG_PI = -Math.log(Math.PI);
+    private final DoubleTensor location;
+    private final DoubleTensor scale;
+
+    public static ContinuousDistribution withParameters(DoubleTensor location, DoubleTensor scale) {
+        return new Cauchy(location, scale);
+    }
+
+    private Cauchy(DoubleTensor location, DoubleTensor scale) {
+        this.location = location;
+        this.scale = scale;
+    }
+
+    @Override
+    public DoubleTensor sample(int[] shape, KeanuRandom random) {
+        if (!scale.greaterThan(0).allTrue()) {
+            throw new IllegalArgumentException("Invalid value for scale. Scale: " + scale.scalar());
```

Comment:
Can we just print out the whole tensor? `throw new IllegalArgumentException("Invalid value for scale. Scale: " + scale);`

Reply:
Do we want to print the object or its contents? To print the Nd4j array, `scale.toString()` might be more appropriate.

Reply:
Yes, I was thinking that we should create a `Tensor#toString` that prints out the first N entries for some constant N.

Reply:
Mmm. For now I'll just print the entire list? (i.e. `.asFlatList())`).